server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -245,6 +247,13 @@ public GroupShardsIterator<ShardIterator> searchShards(
                 preference = Preference.PRIMARY.type();
             }
 
+            if (FeatureFlags.isEnabled(FeatureFlags.TIERED_REMOTE_INDEX)
+                && IndexModule.DataLocalityType.PARTIAL.name()
+                    .equals(indexMetadataForShard.getSettings().get(IndexModule.INDEX_STORE_LOCALITY_SETTING.getKey()))
+                && (preference == null || preference.isEmpty())) {
+                preference = Preference.PRIMARY_FIRST.type();
```

Comment:
For RemoteSnapshot index we are using `Preference.PRIMARY`, how is that different than `Preference.PRIMARY_FIRST`?

Reply:
`Preference.PRIMARY_FIRST` would mean that traffic would be routed to replicas only when primaries are not available to serve the request. So, in case of a node failure and in the interim when the replica is being promoted to primary - the search request would be served by the replica instead of getting failed in case of `Preference.PRIMARY`.